proto/inbound.proto

```diff
@@ -165,3 +177,60 @@ message Identity { string name = 1; }
 // An empty list is equivalent to `.` (matching all names); the list `["foo",
 // "bar"]` is equivalent to "foo.bar." (matching `*.foo.bar`), etc.
 message IdentitySuffix { repeated string parts = 1; }
+
+// Inbound-specific HTTP route configuration (based on the [Gateway API][api]).
+//
+// [api]: https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRoute
+message HttpRoute {
+  io.linkerd.proxy.meta.Metadata metadata = 1;
+
+  // If empty, the host value is ignored.
+  repeated io.linkerd.proxy.http_route.HostMatch hosts = 2;
+
+  // The server MUST return at least one authorization, otherwise all requests
+  // to this route will fail with an unauthorized response.
+  repeated Authz authorizations = 3;
```

Comment:
Perhaps we should instead stipulate that this is combined with a server's authorizations (i.e. by the client) so that we don't have to redundantly serialize the same server authorizations over and over? @adleong